migrations/20190126123621-update-aspect-rows-valueType.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+const validateAspectStatusRanges = require('../db/helpers/aspectUtils')
+                                    .validateAspectStatusRanges;
+const Aspect = require('../db/index').Aspect;
+
+module.exports = {
+  up(qi /* , Sequelize */) {
+    /* If valueType is BOOLEAN or PERCENT and the current ranges wouldn't
+     validate with new rules, change the valueType to NUMERIC.
+     If valueType is NUMERIC, update invalid range with MIN and MAX NUMBER
+     wherever applicable. */
+    return qi.sequelize.transaction(() => {
+      const aspQuery = 'SELECT * from "Aspects"';
+      return qi.sequelize.query(aspQuery, {
+        type: qi.sequelize.QueryTypes.SELECT,
+      }).then((aspects) => {
+        const promisesArr = [];
+        aspects.forEach((asp) => {
+          try {
+            validateAspectStatusRanges(asp);
+          } catch (err) {
+            // if numeric, the only possible exception will be invalid range
+            if (asp.valueType === 'NUMERIC') {
+              const statusRangeMap = new Map();
+              statusRangeMap.set('criticalRange', asp.criticalRange);
+              statusRangeMap.set('warningRange', asp.warningRange);
+              statusRangeMap.set('infoRange', asp.infoRange);
+              statusRangeMap.set('okRange', asp.okRange);
+              const updateObj = {};
+              for (const [key, value] of statusRangeMap.entries()) {
+                if (value && value[0] < Number.MIN_SAFE_INTEGER) {
```

Comment:
isn't it theoretically possible to have both the start and end values be less than the min safe int? if that was the case, we'd still have an invalid range, because we're just correcting the start value here
same if both start and end were greater than max safe int value